tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -245,7 +245,7 @@ final class ScalaJSClassEmitter private (
       js.Block(
           inheritedCtorDef,
           js.Assign(typeVar.prototype, js.New(inheritedCtorRef, Nil)),
-          genAddToPrototype(className, js.Ident("constructor"), typeVar)
+          genAddToPrototype(className, js.StringLiteral("constructor"), typeVar)
```

Comment:
We also rely on the `constructor` property as dot select in the [`scalajsenv`][1] (for `TypedArray` backing of `scala.Array`). IIUC this case could break as well. Maybe we should add `constructor` to the externs?

[1]: https://github.com/scala-js/scala-js/blob/master/tools/scalajsenv.js#L851